java/arcs/android/devtools/ModelUpdateMessage.kt

```diff
@@ -0,0 +1,96 @@
+package arcs.android.devtools
+
+import arcs.android.devtools.DevToolsMessage.Companion.MODEL_UPDATE_MESSAGE
+import arcs.android.devtools.DevToolsMessage.Companion.STORE_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.VERSIONMAP
+import arcs.core.common.ReferenceId
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtEntity
+import arcs.core.crdt.CrdtOperation
+import arcs.core.crdt.CrdtSet
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.data.FieldName
+import arcs.core.storage.ProxyMessage
+import arcs.core.util.JsonValue
+
+/**
+ * An implementation of [StoreMessage] to inform DevTools that a [Store] received a
+ * [ModelUpdate].
+ */
+class ModelUpdateMessage(
+    private val actualMessage: ProxyMessage.ModelUpdate<CrdtData, CrdtOperation, Any?>,
+    private val storeType: String
+) : StoreMessage {
+    override val kind: String = MODEL_UPDATE_MESSAGE
+    override val message: JsonValue<*>
+        get() = JsonValue.JsonObject(
+            "model" to getModel(actualMessage.model),
+            STORE_TYPE to JsonValue.JsonString(storeType)
+        )
+
+    /**
+     * Transform the [CrdtData] into JSON.
+     */
+    private fun getModel(model: CrdtData): JsonValue<*> {
+        when (model) {
+            is CrdtEntity.Data -> {
+                return JsonValue.JsonObject(
+                    VERSIONMAP to model.versionMap.toJson(),
+                    "singletons" to singletonsJson(model.singletons),
+                    "collections" to collectionsJson(model.collections)
+                )
+            }
+            // TODO(heimlich): other types
+        }
+        return JsonValue.JsonNull
```

Comment:
Using the when as an expression here would also help a bit:
```suggestion
        return when (model) {
            is CrdtEntity.Data ->
                JsonValue.JsonObject(
                    VERSIONMAP to model.versionMap.toJson(),
                    "singletons" to singletonsJson(model.singletons),
                    "collections" to collectionsJson(model.collections)
                )
            // TODO(heimlich): other types
            else -> JsonValue.JsonNull
        }
```